tests/pytests/scenarios/reauth/test_reauth.py

```diff
@@ -0,0 +1,50 @@
+import logging
+import os
+import threading
+import time
+
+import pytest
+
+pytestmark = [
+    pytest.mark.slow_test,
+    pytest.mark.windows_whitelisted,
+]
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="module")
+def timeout():
+    return os.environ.get("SALT_CI_REAUTH_MASTER_WAIT", 100)
+
+
+def test_reauth(salt_cli, salt_minion, salt_master, timeout, event_listener):
+    # Make sure they can communicate
+    assert salt_cli.run("test.ping", minion_tgt=salt_minion.id).data is True
+    # Stop the master and minion
+    salt_master.terminate()
```

Comment:
```suggestion
    with salt_master.stopped():
        # daemon stopped here
    # daemon running here
```
This is a fyi, so that you know this exists.